src/Input/TextField/TextField.test.tsx

```diff
@@ -239,3 +239,52 @@ describe('when a non-empty value is passed to', () => {
     });
   });
 });
+
+describe('when allowClear is true', () => {
+  it('should not show the clear icon when value is empty', async () => {
+    const { queryByTestId } = render(
+      <TextField
+        label="label"
+        type="text"
+        allowClear={true}
+        onChange={jest.fn()}
+      />
+    );
+
+    const clearButton = queryByTestId('clear-button');
+    expect(clearButton).not.toBeInTheDocument();
+  });
+
+  it('should show the clear icon when value is not empty', () => {
+    const { queryByTestId } = render(
+      <TextField
+        label="label"
+        type="text"
+        value="123"
+        allowClear={true}
+        onChange={jest.fn()}
+      />
+    );
+
+    const clearButton = queryByTestId('clear-button');
+    expect(clearButton).toBeInTheDocument();
+  });
+
+  it('should clear the input value and trigger onChange when click clear button', () => {
+    const onChange = jest.fn();
+
+    const { queryByTestId } = render(
+      <TextField
+        label="label"
+        type="text"
+        value="123"
+        allowClear={true}
+        onChange={onChange}
+      />
+    );
+
+    const clearButton = queryByTestId('clear-button');
+    userEvent.click(clearButton);
+    expect(onChange).toHaveBeenCalledTimes(1);
+  });
+});
```

Comment:
Do you think you can add a test case for when `type="password"` and `allowClear={true}`. Current behavior in that case is that only the password icon is shown, which is good imo. But we should make that behavior explicit.